hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -367,9 +368,11 @@ public static DBCheckpoint createOmSnapshotCheckpoint(
       dbCheckpoint = store.getSnapshot(snapshotInfo.getCheckpointDirName());
       // Clean up active DB's deletedTable right after checkpoint is taken,
       // with table write lock held
-      deleteKeysInSnapshotScopeFromDTableInternal(omMetadataManager,
+      deleteKeysFromDelKeyTableInSnapshotScope(omMetadataManager,
+          snapshotInfo.getVolumeName(), snapshotInfo.getBucketName());
+      // Clean up deletedDirectoryTable as well
+      deleteKeysFromDelDirTableInSnapshotScope(omMetadataManager,
```

Comment:
I intended to share the `DELETED_TABLE` lock. Now that you bring this up we might need to add a new lock because of `DirectoryDeletingService`. Will take a look.

Reply:
Please also keep https://issues.apache.org/jira/browse/HDDS-8067 in context.

Reply:
@aswinshakil I decided to just add a TODO in this PR so I can review the table lock situation comprehensively in [HDDS-8067](https://issues.apache.org/jira/browse/HDDS-8067). Note `getTableLock()` is currently only used here in this method. We will most likely need to grab this lock in KDT and DDT as well when accessing those tables.